internal/route/websocket.go

```diff
@@ -0,0 +1,113 @@
+package route
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+)
+
+// Message that will be sent to client
+type Message struct {
+	Event string `json:"event"`
+	Data  struct {
+		ListID string `json:"list_id"`
+		TodoID string `json:"todo_id"`
+		Done   bool   `json:"done"`
+	} `json:"data"`
+}
+
+// RegisterMessage registers client
+type RegisterMessage struct {
+	Message Message
+	Client  *websocket.Conn
+}
+
+type socketManager struct {
+	clients   map[string][]*websocket.Conn
```

Comment:
You'll probably want to use a `sync.Mutex` here. Your `clients` map is written to from separate goroutines. This works fine when you're testing it out, but with multiple users you'll run into a race condition.

By adding `m sync.Mutex` to this struct it'll be good to go. The "zero-value" is fine.